Framework/Source/Graphics/Model/Loaders/AssimpModelImporter.cpp

```diff
@@ -555,11 +555,21 @@ namespace Falcor
         return loader.initModel(filename);
     }
 
+    bool AssimpModelImporter::isUsedNode(const aiNode* pNode) const
+    {
+        return (mBoneNameToIdMap.count(pNode->mName.C_Str()) > 0) || (mAdditionalUsedNodes.count(pNode) > 0);
+    }
+
     uint32_t AssimpModelImporter::initBone(const aiNode* pCurNode, uint32_t parentID, uint32_t boneID)
     {
-        assert(mBoneNameToIdMap.find(pCurNode->mName.C_Str()) != mBoneNameToIdMap.end());
+        assert(isUsedNode(pCurNode));
         assert(pCurNode->mNumMeshes == 0);
-        mBoneNameToIdMap[pCurNode->mName.C_Str()] = boneID;
+
+        auto it = mBoneNameToIdMap.find(pCurNode->mName.C_Str());
+        if (it != mBoneNameToIdMap.end())
```

Comment:
When does this thing happens? Aren't we overriding `second` with a new `boneID`?

Reply:
It's the bone initializer, it's only overriding -1 with the actual array index. It used to set unconditionally, now it checks whether it's a valid bone name (instead of some other node) before setting.